execution/executor/src/block_processor.rs

```diff
@@ -567,7 +569,7 @@ where
         transactions: &[Transaction],
         vm_outputs: Vec<TransactionOutput>,
         parent_trees: &ExecutedTrees,
-    ) -> Result<ProcessedVMOutput> {
+    ) -> Result<(ProcessedVMOutput, Option<ValidatorSet>)> {
```

Comment:
Was wondering if we need to change the VM api to return that `Option<ValidatorSet>` so that we don't need to worry about parsing logic in the executor code?

Reply:
We could easily put that logic there; it's more of a question of who should know about validator sets, magic events, and magic access paths. My philosophy was: 
- Keep the VM as dumb and Libra-agnostic as possible: it just takes a transaction and evaluates it to a write set + events + status
- Put all of the Libra-specific logic in the executor. It already has to know about things like the validator set, so it makes sense for it to now about the magic validator set event.